crate.go

```diff
@@ -36,31 +36,58 @@ type crateReadResponse struct {
 }
 
 type crateEndpoint struct {
-	pool     *pgx.ConnPool
-	poolConf pgx.ConnPoolConfig
+	pool     *pgxpool.Pool
+	poolConf *pgxpool.Config
 }
 
 func newCrateEndpoint(ep *endpointConfig) *crateEndpoint {
-	connConf := pgx.ConnConfig{
-		Host:     ep.Host,
-		Port:     ep.Port,
-		User:     ep.User,
-		Password: ep.Password,
-		Database: ep.Schema,
-		Dial: (&net.Dialer{
-			KeepAlive: 30 * time.Second,
-			Timeout:   time.Duration(ep.ConnectTimeout) * time.Second,
-		}).Dial,
+
+	// pgx4 starts using connection strings exclusively, in both URL and DSN formats.
+	// The single entrypoint to obtain a valid configuration object, is `pgx.ParseConfig`,
+	// which aims to be compatible with libpq.
+
+	// ParseConfig builds a *Config from connString with similar behavior to the PostgreSQL
+	// standard C library libpq. It uses the same defaults as libpq (e.g. port=5432), and
+	// understands most PG* environment variables.
+	//
+	// ParseConfig closely matches the parsing behavior of libpq. connString may either be
+	// in URL or DSN format. connString also may be empty to only read from the environment.
+	// If a password is not supplied it will attempt to read the .pgpass file.
+	//
+	// See https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING for details.
+	//
+	//   # Example DSN
+	//   user=jack password=secret host=pg.example.com port=5432 dbname=mydb sslmode=verify-ca
+	//
+	//   # Example URL
+	//   postgres://jack:secret@pg.example.com:5432/mydb?sslmode=verify-ca
+	connectionString := fmt.Sprintf(
+		"postgres://%s:%s@%s:%v/%s?connect_timeout=%v&pool_max_conns=%v",
+		ep.User, ep.Password, ep.Host, ep.Port, ep.Schema, ep.ConnectTimeout, ep.MaxConnections)
```

Comment:
I think a password in this context may need to be URL encoded for special characters to work.

Reply:
Good point!

a) Do you suggest to encode it, or do you suggest to use the DSN notation instead?
b) FYI: In a subsequent path, I would like to add functionality to obtain the connection string from the configuration file 1:1, and document all other options supported by pgx4 out of the box.

Reply:
Hi again. Later down the line, GH-82 will start using DSN-style connection strings instead.